docs/pages/access-controls/reference.mdx

```diff
@@ -496,24 +496,49 @@ sign-on provider.
 |`{{internal.windows_logins}}`|List of allowed Windows logins for the user.|
 |`{{external.xyz}}` | Substituted with a value from an external [SSO provider](https://en.wikipedia.org/wiki/Single_sign-on). If using SAML, this will be expanded with "xyz" assertion value. For OIDC, the variable is expanded to the value of the "xyz" claim. See the next section for more information on referring to the `external` variable in Teleport roles. |
 
-#### Accessing external traits in Teleport roles
+#### Referring to internal traits in Teleport roles
 
+The `internal` trait namespace includes only the exact internal trait names
+included in the above table.
+They can be set in the `spec.traits` field of the
+[user resource](../reference/resources.mdx#user), or passed from an external
+IdP.
+
+Internal traits may be referenced in built-in roles such as the `access` role,
+and they may be different when a user is accessing a leaf cluster vs a root
+cluster.
+For example, when accessing a leaf cluster, the `logins` trait will be set to
+the complete list of SSH logins allowed for the user, which may include values
+from the user's `logins` trait in the root cluster as well as logins hardcoded
+in roles the user holds in the root cluster.
```

Comment:
Instead of using "may", I would be more specific in spelling out the logic for assigning internal traits in leaf clusters and root clusters. Otherwise, this logic sounds pretty complicated and difficult to follow.

Reply:
the logic *is* complicated and difficult to follow. I expanded this explanation, listed all affected traits, and pulled it into a warning admonition